lightning/src/ln/peers/handler.rs

```diff
@@ -0,0 +1,3280 @@
+	fn peers_needing_send<'a>(&'a mut self) -> Filter<IterMut<'a, Descriptor, Peer<TransportImpl>>, fn(&(&'a Descriptor, &'a mut Peer<TransportImpl>)) -> bool> {
+		self.peers.iter_mut().filter(|(_, peer)| {
+			let has_outbound_sync = match &peer.post_init_state {
+				None => false,
+				Some(post_init_state) => match &post_init_state.sync_status {
+					InitSyncTracker::NoSyncRequested => false,
+					InitSyncTracker::ChannelsSyncing(_) => true,
+					InitSyncTracker::NodesSyncing(_) => true,
+				}
+			};
+
+			has_outbound_sync || !peer.outbound_queue.is_empty()
+		})
+	}
+}
+
+#[cfg(not(any(target_pointer_width = "32", target_pointer_width = "64")))]
+fn _check_usize_is_32_or_64() {
+	// See below, less than 32 bit pointers may be unsafe here!
+	unsafe { mem::transmute::<*const usize, [u8; 4]>(panic!()); }
+}
+
+/// SimpleArcPeerManager is useful when you need a PeerManager with a static lifetime, e.g.
+/// when you're using lightning-net-tokio (since tokio::spawn requires parameters with static
+/// lifetimes). Other times you can afford a reference, which is more efficient, in which case
+/// SimpleRefPeerManager is the more appropriate type. Defining these type aliases prevents
+/// issues such as overly long function definitions.
+pub type SimpleArcPeerManager<SD, M, T, F, C, L> = Arc<PeerManager<SD, SimpleArcChannelManager<M, T, F, L>, Arc<NetGraphMsgHandler<Arc<C>, Arc<L>>>, Arc<L>>>;
+
+/// SimpleRefPeerManager is a type alias for a PeerManager reference, and is the reference
+/// counterpart to the SimpleArcPeerManager type alias. Use this type by default when you don't
+/// need a PeerManager with a static lifetime. You'll need a static lifetime in cases such as
+/// usage of lightning-net-tokio (since tokio::spawn requires parameters with static lifetimes).
+/// But if this is not necessary, using a reference is more efficient. Defining these type aliases
+/// helps with issues such as long function definitions.
+pub type SimpleRefPeerManager<'a, 'b, 'c, 'd, 'e, 'f, 'g, SD, M, T, F, C, L> = PeerManager<SD, SimpleRefChannelManager<'a, 'b, 'c, 'd, 'e, M, T, F, L>, &'e NetGraphMsgHandler<&'g C, &'f L>, &'f L>;
+
+/// A PeerManager manages a set of peers, described by their SocketDescriptor and marshalls socket
+/// events into messages which it passes on to its MessageHandlers.
+///
+/// Rather than using a plain PeerManager, it is preferable to use either a SimpleArcPeerManager
+/// a SimpleRefPeerManager, for conciseness. See their documentation for more details, but
+/// essentially you should default to using a SimpleRefPeerManager, and use a
+/// SimpleArcPeerManager when you require a PeerManager with a static lifetime, such as when
+/// you're using lightning-net-tokio.
+pub struct PeerManager<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref> where
+	CM::Target: ChannelMessageHandler,
+	RM::Target: RoutingMessageHandler,
+	L::Target: Logger {
+		inner: PeerManagerImpl<Descriptor, CM, RM, L>,
```

Comment:
Using default type parameters in the public interface requires that the `Transport` type has public visibility as well. 

This would mean leaking private types as well as making it unclear from a user perspective on whether or not they needed to implement a Transport object (it would show up in the rustdoc as well). Since this is just a pattern for testability, I chose to hide it as an inner module which allowed the use of dependency injection without an interface change to the user.